Add section on how to check types on CI (#873)
withastro/docs: src/pages/en/guides/typescript.md

<h2>{greeting}, {name}!</h2>
```

## Type checking

> To see type errors in your editor please make sure that you have [Astro extention](/en/editor-setup/). Please note that `astro start` and `astro build` will not run any type checking, they just transpile the code with esbuild. To make sure that typescript errors prevent your code from building you can change your "build" script in `package.json`:

Reviewer: ```suggestion
To see type errors in your editor please make sure that you have the [Astro VS Code extention](/en/editor-setup/) installed. Please note that the `astro start` and `astro build` commands will transpile the code with esbuild, but will not run any type checking. To prevent your code from building if it contains TypeScript errors, change your "build" script in `package.json` to the following:
```
Just tidied up some of the language to match our current style!

> To see type errors in your editor please make sure that you have the [Astro VS Code extention](/en/editor-setup/) installed. Please note that the `astro start` and `astro build` commands will transpile the code with esbuild, but will not run any type checking. To prevent your code from building if it contains TypeScript errors, change your "build" script in `package.json` to the following:

Reviewer: ```suggestion
To see type errors in your editor, please make sure that you have the [Astro VS Code extention](/en/editor-setup/) installed. Please note that the `astro start` and `astro build` commands will transpile the code with esbuild, but will not run any type checking. To prevent your code from building if it contains TypeScript errors, change your "build" script in `package.json` to the following:
```
Oops, spotted a missing comma! I'm just going to clean that up myself so we can merge this.  :)

```diff
-    "build": "astro build",
+    "build": "astro check && tsc --noEmit && astro build",
```

:::note

> `astro check` command only checks types within `.astro` files, and `tsc --noEmit` only checks types within `.ts` and `.tsx` files.

Reviewer: ```suggestion
`astro check` only checks types within `.astro` files, and `tsc --noEmit` only checks types within `.ts` and `.tsx` files.
```
removed "command" because we can be briefer in the note, and both of these commands should be referenced the same way to be consistent. We did mention "commands" in the paragrah above, so I think we can just be shorter, and keep them the same, here.
